### PR TITLE
18245: public subnet filtering gateway

### PR DESCRIPTION
**Pull request by CyrusJavan**
This PR adds the ability to create and manage
a public subnet filtering gateway.

We do this by adding some additional options
to the gateway resource. Still to do is adding
support for these new options in TF Export.

### CODE REVIEW

**Comment by inc1t3Ful** [comment]
Docs are good 👍  I think Tim wanted to re-discuss implementation choices at some point

File: aviatrix/resource_aviatrix_gateway.go [hunk 1]
```diff
@@ -12,6 +12,51 @@ import (
 	"github.com/terraform-providers/terraform-provider-aviatrix/goaviatrix"
 )
 
+var invalidPublicSubnetFilteringGatewayConfigKeys = []string{
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_gateway.go]: Move "invalidPublicSubnetFilteringGatewayConfigKeys" to the bottom.

File: aviatrix/resource_aviatrix_gateway.go [hunk 2]
```diff
@@ -412,6 +456,34 @@ func resourceAviatrixGateway() *schema.Resource {
 				Computed:    true,
 				Description: "FQDN gateway lan interface id.",
 			},
+			"public_subnet_filtering": {
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_gateway.go]: Maybe change to "enable_public_subnet_filtering" or "public_subnet_filtering_enabled"?

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]
Probably need to change "public_subnet_filtering" to "enable_public_subnet_filtering" or "public_subnet_filtering_enabled".

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**